pkgs/tools/misc/idutils/default.nix

```diff
@@ -1,27 +1,31 @@
-{ fetchurl, lib, stdenv, emacs }:
+{ fetchurl, lib, stdenv, emacs, gnulib, autoconf, bison, automake, gettext, gperf, texinfo, perl, rsync, gawk }:
 
 stdenv.mkDerivation rec {
-  name = "idutils-4.6";
+  pname = "idutils";
+  version = "4.6";
 
   src = fetchurl {
-    url = "mirror://gnu/idutils/${name}.tar.xz";
+    url = "mirror://gnu/idutils/${pname}-${version}.tar.xz";
     sha256 = "1hmai3422iaqnp34kkzxdnywl7n7pvlxp11vrw66ybxn9wxg90c1";
   };
 
   preConfigure =
     ''
-       # Fix for building on Glibc 2.16.  Won't be needed once the
-       # gnulib in idutils is updated.
-       sed -i '/gets is a security hole/d' lib/stdio.in.h
+    # replace embedded gnulib tests with those from gnulib package
+    shopt -s extglob
+    (cd gnulib-tests; rm -r !(Makefile.am))
+    shopt -u extglob
```

Comment:
```suggestion
    (shopt -s extglob; cd gnulib-tests; rm -r !(Makefile.am))
```
Does that work?

Reply:
I agree this would be better, but I can't get it to work.  extglob disables subsequent shell history !(..) expansion, but history expansion takes place very early: before shopt can enable extglob. Thus extglob must be enabled in a previous command or outer context. This would work:
   bash -O extglob -c "cd gnulib-tests; rm -r !(Makefile.am)"